paddle/framework/prune.cc

```diff
@@ -106,5 +108,26 @@ void Prune(const ProgramDesc& input, ProgramDesc* output) {
   prune_impl(input, output, 0);
 }
 
+void inference_optimize_impl(const ProgramDesc& input, ProgramDesc* output,
```

Comment:
The logic of `inference_optimize_impl` is quite simple. Maybe we can implement it in Python.